lightning/lightning.go

```diff
@@ -229,17 +235,39 @@ func (l *Lightning) run(taskCfg *config.Config) (err error) {
 		web.BroadcastEndTask(err)
 	}()
 
-	failpoint.Inject("SkipRunTask", func() error {
+	failpoint.Inject("SkipRunTask", func() {
 		if recorder, ok := l.ctx.Value(&taskCfgRecorderKey).(chan *config.Config); ok {
 			select {
 			case recorder <- taskCfg:
 			case <-ctx.Done():
-				return ctx.Err()
+				failpoint.Return(ctx.Err())
 			}
 		}
-		return nil
+		failpoint.Return(nil)
 	})
 
+	if err := taskCfg.TiDB.Security.RegisterMySQL(); err != nil {
```

Comment:
Are there any problems if we run multi lightning tasks with in the same tidb? Or tidb will refuse to run more than one lightning tasks?

Reply:
currently TiDB will only run one task, and `RunOnce` is blocked thus it's unnatural to concurrently call `RunOnce` on one lightning (I hope). 

For multiple lightning in one process, `RegisterMySQL` may fail because `tlsConfigRegistry` is a global variable, I'll be careful for that